### PR TITLE
src/doc/en/developer/review.rst: Update section 'Release Process'

### PR DESCRIPTION
**Pull request by mkoeppe**
Updating the description of the release process to match the current practice.

As noted in https://groups.google.com/g/sage-devel/c/uWHaWK7b6H4/m/cM77D8E5AAAJ

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/doc/en/developer/review.rst [hunk 1]
```diff
@@ -180,14 +180,15 @@ Sage Release Manager uses to make releases. Here it is as of 2023:
 **Beta Release Stage**: For preparing a new beta release or the first release
 candidate, all positively reviewed PRs with the forthcoming release
 milestone are considered. PRs that have dependencies not merged yet are ignored.
-The Release Manager merges PRs in batches of 10 to 20 PRs, taking the
+The Release Manager merges PRs in batches of 10 to 20 PRs, without taking the
 PR priority into account. If a merge conflict of a PR to the Release
 Manager's branch occurs, the PR is set back to "needs work" status by the
```
Comment (kwankyu) [inline src/doc/en/developer/review.rst]: If this is true (@vbraun ?), then the "without ..." clause, which sounds negative, may just be deleted.

**Comment by kwankyu** [comment]
Otherwise, lgtm.

File: src/doc/en/developer/review.rst (hunk 1 above)
Comment (kwankyu) [reply src/doc/en/developer/review.rst]: The release management script does not consider PR priority. Anyway the "without ..." clause seems redundant.

Comment (kwankyu) [reply src/doc/en/developer/review.rst]: Thanks. I also suspected "PRs that have dependencies not merged yet are ignored." is not true. Perhaps only reviewers care "dependencies".

Would you reformat the lines around 182-184?

Comment (kwankyu) [reply src/doc/en/developer/review.rst]: Thanks.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**